src/main/kotlin/ecommerce/auth/AuthorizationExtractor.kt

```diff
@@ -0,0 +1,11 @@
+package ecommerce.auth
+
+import jakarta.servlet.http.HttpServletRequest
+
+interface AuthorizationExtractor<T> {
+    fun extract(request: HttpServletRequest): T
+
+    companion object {
+        const val AUTHORIZATION = "Authorization"
+    }
+}
```

Comment:
Since there is currently a single implementation of this interface, I'm curious about the reason why you chose to keep this as an interface! It's not wrong, but some developers like me actually prefer to make an interface when we actually need it.